pkg/cloudfoundry/ManifestUtils.go

```diff
@@ -0,0 +1,226 @@
+package cloudfoundry
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"reflect"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+const propApplications = "applications"
+const propBuildpacks = "buildpacks"
+const propBuildpack = "buildpack"
+
+// Manifest ...
+type Manifest struct {
+	self     map[string]interface{}
+	modified bool
+	name     string
+}
+
+var m Manifest
+
+var _readFile = ioutil.ReadFile
+
+// ReadManifest ...
+func ReadManifest(name string) (Manifest, error) {
+
+	log.Entry().Infof("Reading manifest file  '%s'", name)
+
+	m := Manifest{self: make(map[string]interface{}), name: name, modified: false}
+
+	content, err := _readFile(name)
+	if err != nil {
+		return m, errors.Wrapf(err, "cannot read file '%v'", m.name)
+	}
+
+	err = yaml.Unmarshal(content, &m.self)
+
+	if err != nil {
+		return m, errors.Wrapf(err, "Cannot parse yaml file '%s': %s", m.name, string(content))
+	}
+
+	log.Entry().Infof("Manifest file '%s' has been parsed", m.name)
+
+	return m, nil
+}
+
+// WriteManifest ...
+func (m *Manifest) WriteManifest() error {
+
+	d, err := yaml.Marshal(&m.self)
+	if err != nil {
+		return err
+	}
+
+	log.Entry().Debugf("Writing manifest file '%s'", m.name)
+	err = ioutil.WriteFile(m.name, d, 0644)
+
+	if err == nil {
+		m.modified = false
+	}
+
+	log.Entry().Debugf("Manifest file '%s' has been written", m.name)
+	return err
+}
+
+// GetName ...
+func (m Manifest) GetName() string {
+	return m.name
+}
+
+// GetApplications ...
+func (m Manifest) GetApplications() ([]interface{}, error) {
+	return toSlice(m.self)
+}
+
+// ApplicationHasProperty Checks if the application denoted by 'index' has the property 'name'
+func (m Manifest) ApplicationHasProperty(index int, name string) (bool, error) {
```

Comment:
Not covered in tests

Reply:
aaahh, yes. I was too focused on transferring the existing coding, and also the tests from groovy to go. I forgot the tests wrt new methods. Of course I will add  these tests ...

Reply:
I used the golang coverage to see this. https://golangdocs.com/code-coverage-in-golang

Reply:
done